### DISCUSSION TITLE
Publish then subscribe and repeat

### DISCUSSION THREAD

**Discussion by AbdullahGaw** [aws/aws-iot-device-sdk-python#327] [Q&A] [answered]
I have a function that publishes to mqtt topic then calls a function where I subscribe to another topic inside a while True loop.

In the subscribe callback function i call the function that publishes to mqtt. If a condition is not me

What i want to accomplish:
be able to publish and wait for a certain response from mqtt then publish again and loop until a condition is met.

what actually happens: 
My script publish to mqtt then mqtt publish to me then I publish to mqtt then nothing happens! And i get subscribe timeout exception 

I tried using one client for both sub and pub and i tried a client for each but neither worked!

**Comment by jmklix** [comment, marked as answer]
Are you trying to subscribe multiple times? You said
> I subscribe to another topic inside a while True loop.

You only need to subscribe once to a specific topic.

**Reply by jmklix** [reply]
Please let us know if you are still having problems subscribing to topics. Marking this as answered for now.

**Reply by AbdullahGaw** [reply]
Hello Thank you! for your response. In my case It was more convenient to use two clients one for publishing and the second for subscribing. This was there is no conflicts. one does not have to wait for the other to complete which is what I needed. Thank you!
